dcpquery/etl/__init__.py

```diff
@@ -65,6 +65,14 @@ def get_file_extension(filename):
     return file_extension
 
 
+def get_schema_type(body):
+    try:
+        regex_search_result = re.search('([^\/]+$)', body['describedBy']) # noqa W605
+        return regex_search_result.group(1)
+    except TypeError:
```

Comment:
What is TypeError guarding against? There are three types of errors possible here: `describedBy` not in body, `describedBy` is not a string, and the regexp doesn't match. It seems that TypeError guards against the second, but not the first or third, right? Also, `None` is returned by default. So consider using this instead:
```
def get_schema_type(body: dict):
    search_result = re.search(r'([^\/]+$)', body.get('describedBy', ''))
    if search_result:
        return search_result.group(1)
```
Also, it looks like you are splitting by `/`, but you are not asserting that there is at least one `/`, so your code is actually equivalent to:
```
def get_schema_type(body: dict):
    return body.get('describedBy', '').split('/')[-1]
```
But is that what you want? Is schema type nullable? What are the circumstances under which this should return None?

If you don't want to document such circumstances, the correct defensive behavior is to assert that `describedBy` *is* in the metadata doc being processed, and that it *is* a string. To be even more strict, I would recommend asserting that it starts with `https://schema.humancellatlas.org/`. So consider using this:
```
def get_schema_type(body: dict):
    assert body['describedBy'].startswith('https://schema.humancellatlas.org/')
    return body['describedBy'].split('/')[-1]
```

Reply:
schema type is nullable, it should return none for the data files that dont have an attached schema type (or a describedBy field) such as data files that we want to represent in the query service as the output of processes. 

The `TypeError` was meant to catch those types of files that dont have a `describedBy` section. From my understanding `describedBy` should always be a string of a specific type so I didnt think that was worth verifying. 

It appears that the `describedBy` url is env specific (and http instead of https) so I would only be able to do
`assert body['describedBy'].startswith('http://schema')`